lib/iris/tests/test_analysis.py

```diff
@@ -1926,18 +1885,21 @@ def test__Groupby_repr():
     assert repr(grouper) == "_Groupby(['year'], shared_coords=['time'])"
 
 
-CUBE = iris.cube.Cube(0)
-
-
 @pytest.mark.parametrize(
     "kwargs,expected",
     [
         ({}, "s"),
         ({"test": "m"}, "s"),
         ({"weights": None}, "s"),
         ({"weights": [1, 2, 3]}, "s"),
-        ({"weights": iris.analysis._Weights([1], CUBE)}, "s"),
-        ({"weights": iris.analysis._Weights([1], CUBE, units="kg")}, "s kg"),
+        ({"_weights_units": None}, "s"),
+        ({"test": "m", "_weights_units": None}, "s"),
+        ({"weights": None, "_weights_units": None}, "s"),
+        ({"weights": [1, 2, 3], "_weights_units": None}, "s"),
+        ({"_weights_units": "kg"}, "s"),
+        ({"test": "m", "_weights_units": "kg"}, "s"),
+        ({"weights": None, "_weights_units": "kg"}, "s"),
+        ({"weights": [1, 2, 3], "_weights_units": "kg"}, "s kg"),
```

Comment:
FYI `parametrize` can do a matrix, if you're looking to do an exhaustive test of all combinations. That wouldn't work with the `expected` element, but since only one of these combinations is expected to return `"s kg"` you could probably put a conditional within the test.

Not a deal breaker for me.

Reply:
True! However, including all test cases (e.g., also the one with an empty dict) would add quite some additional code, so I would prefer to keep it like that in terms of readability.